### PR TITLE
Refactor code utilising classes

### PR DESCRIPTION
**Pull request by omushpapa**
The code is refactored to utilises classes. Global variables removed. Added icon. Use clam theme. Clear display when a new operation takes place (no need to press AC). Reformat docstrings

### CODE REVIEW

**Comment by tasdikrahman** [comment]
Hey @giantas 

Awesome work!

Some comments, 
- I am actually not able to get this up and running, which system are you on? Did you test it properly? I tried with both python2 and 3

## `python3`

```python

$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 31, in __init__
    self.icon = tk.PhotoImage(data=icon_data)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tkinter/__init__.py", line 3539, in __init__
    Image.__init__(self, 'photo', name, cnf, master, **kw)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tkinter/__init__.py", line 3495, in __init__
    self.tk.call(('image', 'create', imgtype, name,) + options)
_tkinter.TclError: couldn't recognize image data
```

## `python2`

```python
$ python main.py
Traceback (most recent call last):
  File "main.py", line 3, in <module>
    from gui import TkGUI
  File "pyCalc/pyCalc/gui.py", line 3, in <module>
    import tkinter as tk
ImportError: No module named tkinter
```
For now I am only concentrating on `python2`, so I changed the above import to `import Tkinter as tk` citing [https://stackoverflow.com/a/25905642/3834059](https://stackoverflow.com/a/25905642/3834059)

Further the import `from Tkinter import ttk` is changed to `import ttk` citing [https://stackoverflow.com/a/23984633/3834059](https://stackoverflow.com/a/23984633/3834059)

I am not sure which version of python you are running. Here are the versions that I am running and tried your PR with

```bash
$ python --version
Python 2.7.10
$ python --version
Python 3.6.0
```

**Comment by tasdikrahman** [comment]
Also, I see that you committed 3 images or so. But they are identical to each other. Any specific reasons for doing so?

**Comment by omushpapa** [comment]
Perhaps the code is not functioning because you are using Python 3.6 yet you originally coded it for Python 3.4 (as indicated by the shebang line in main.py). The images are just but different resolutions. I figured they might be of use to anyone in the case of symbolic links, etc.

**Comment by omushpapa** [comment]
`so I changed the above import to import Tkinter as tk citing https://stackoverflow.com/a/25905642/3834059` this change has not been reflected in the code. It still reads `import tkinter as tk` [check](https://github.com/prodicus/pyCalc/blob/master/pyCalc/main.py)

**Comment by tasdikrahman** [comment]
Hey @giantas 

> so I changed the above import to import Tkinter as tk citing https://stackoverflow.com/a/25905642/3834059 this change has not been reflected in the code. It still reads import tkinter as tk check

I have not pushed anything to the master branch if you are asking that, was trying the changes you made locally on my dev box. 

I removed the import errors by doing this

```python
$ head -n 10 gui.py
#!/usr/bin/env python

try:
	import Tkinter as tk
	import ttk
except ImportError:
	import tkinter as tk  # python3
	from tkinter import ttk
import parser
import base64
```

After that, the traceback for the errors reduce to 

## python3

```python
$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 35, in __init__
    self.icon = tk.PhotoImage(data=icon_data)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tkinter/__init__.py", line 3539, in __init__
    Image.__init__(self, 'photo', name, cnf, master, **kw)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tkinter/__init__.py", line 3495, in __init__
    self.tk.call(('image', 'create', imgtype, name,) + options)
_tkinter.TclError: couldn't recognize image data
```

Says, can't recognize image data. did you face this issue?

## python2

```python
$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 25, in __init__
    super(TkGUI, self).__init__()
TypeError: must be type, not classobj
```

**Comment by omushpapa** [comment]
`Says, can't recognize image data. did you face this issue?` must be a Python 3.6 issue. I may not be able to resolve this since I cannot install multiple Python3 versions in my OS. But let me try implementing the same in Python two and see if I will experiencce the same error. I am hoping you are using Python 2.7.

**Comment by tasdikrahman** [comment]
> I may not be able to resolve this since I cannot install multiple Python3 versions in my OS

Take a look at [this](http://python-guide-pt-br.readthedocs.io/en/latest/dev/virtualenvs/) for solving that 

>  I am hoping you are using Python 2.7.

Yes. Python 2.7.10 to be specific. I am testing on both, but we can focus on one version of python for now.

**Comment by tasdikrahman** [comment]
We should first focus on the `python2` issue, see if you can replicate this error traceback 

```python
$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 25, in __init__
    super(TkGUI, self).__init__()
TypeError: must be type, not classobj
```

I have mentioned my python2 version in the thread

**Comment by omushpapa** [comment]
Ok on the Python 2 fix. Will get back to you once I get a fix.

**Comment by omushpapa** [comment]
Just got a fix for it. Perhaps I should create another pull request?

**Comment by omushpapa** [comment]
The fix is summarily mentioned [here](https://stackoverflow.com/questions/43767988/typeerror-super-argument-1-must-be-type-not-classobj)

**Comment by omushpapa** [comment]
No need for another request, you can just merge the changes

File: pyCalc/gui.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
 #! /usr/bin/env python3.4
```
Comment (tasdikrahman) [inline pyCalc/gui.py]: we can change this to `#!/usr/bin/env python`

**Comment by tasdikrahman** [comment]
Hey @giantas 

```python
$ python --version
Python 2.7.10

$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 42, in __init__
    self.icon = tk.PhotoImage(data=icon_data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk/Tkinter.py", line 3366, in __init__
    Image.__init__(self, 'photo', name, cnf, master, **kw)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk/Tkinter.py", line 3320, in __init__
    self.tk.call(('image', 'create', imgtype, name,) + options)
_tkinter.TclError: couldn't recognize image data
```

File: pyCalc/gui.py (hunk 1 above)
Comment (tasdikrahman) [reply pyCalc/gui.py]: as well as in `gui.py`

**Comment by omushpapa** [comment]
I do not understand why it didn't work. Perhaps you could try encoding a 16x16 version of the icon by yourself [here](http://www.motobit.com/util/base64-decoder-encoder.asp). Then ensure the encoding is utf-8. Thing is, I've used that method on my pc with different apps. I'll check it out in the morning using the PCs of my colleagues. 

Sorry, I left out the shebang change.

**Comment by omushpapa** [comment]
Apparently, this error was not visible on Linux, had to use a Window system. The problem was that I encoded a png file. I fixed it by encoding a gif file. 

This should fix the same error on Python3.6

```$ python --version
Python 2.7.10

$ python main.py
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    app = TkGUI()
  File "pyCalc/pyCalc/gui.py", line 42, in __init__
    self.icon = tk.PhotoImage(data=icon_data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk/Tkinter.py", line 3366, in __init__
    Image.__init__(self, 'photo', name, cnf, master, **kw)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk/Tkinter.py", line 3320, in __init__
    self.tk.call(('image', 'create', imgtype, name,) + options)
_tkinter.TclError: couldn't recognize image data```

**Comment by tasdikrahman** [comment]
Hey @giantas 

Awesome work! 

Works now on both python2 and python3.  Please update the README to reflect the changelog and add yourself as a contributor :)

Also, if the gif in README is outdated. You can update that too if you feel like it

Great work!

**Pull request merged**